### PR TITLE
Added Quality Assurance and testing Quiz Questions

### PR DESCRIPTION
**Pull request by SWASTI-SUNDAR**
Added Quality Assurance and testing Quiz Questions
 #102 

- [ ] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [ ] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [ ] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [ ] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [ ] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/general-cs-quiz.ts [hunk 1]
```diff
@@ -1597,6 +1597,67 @@ const generalCSQuiz = [
     Explanation:
       "Floyd's Tortoise-Hare Algorithm is for finding a circular Linked list. The Hare is a pointer which jumps 2 nodes while the Tortoise moves one node at a time. In case they both meet, we get a circular linked list. If any of them becomes nullptr, then there is no Circular Linked list.",
     Link: "https://en.wikipedia.org/wiki/Cycle_detection"
+  },
+  {
+    Question:
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1602]: Please remove all of these changes in this file since it was already merged in

File: src/data/linux-quiz.ts [hunk 2]
```diff
@@ -1330,6 +1330,54 @@ const linuxQuiz = [
     Distractor3: "stop [signal] PID",
     Explanation: "The kill command is used to stop a running process.",
     Link: "https://www.linuxfoundation.org/blog/blog/classic-sysadmin-how-to-kill-a-process-from-the-command-line"
+  },
+  {
+    Question:
```
Comment (jdwilkin4) [inline src/data/linux-quiz.ts:1335]: Please remove all of these changes in this file since it is already in another PR

File: src/data/sql-quiz.ts [hunk 3]
```diff
@@ -831,20 +831,22 @@ const sqlQuiz = [
     Link: "https://www.freecodecamp.org/news/a-beginners-guide-to-sql/"
   },
   {
-    Question:
-      "What SQL statement contains a valid subquery?",
-    Answer: "SELECT * FROM employees WHERE wage > (SELECT AVG(wage) FROM employees)",
+    Question: "What SQL statement contains a valid subquery?",
```
Comment (jdwilkin4) [inline src/data/sql-quiz.ts:834]: Please remove all of these changes in this file since it is already in another PR

File: src/data/quality-assurance-quiz.ts [hunk 4]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
     Explanation:
       "A good QA engineer should understand the full life cycle of software products in order to effectively plan and test them. There are 7 phases of the Software Development Life Cycle",
     Link: "https://www.freecodecamp.org/news/how-to-become-a-quality-assurance-engineer-qa-engineer-career-guide/"
+  },
+
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:186]: remove this extra unnecessary space please 

```suggestion
```

File: src/data/quality-assurance-quiz.ts [hunk 5]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+  {
+    Question:
+      "What is the difference between 'functional testing' and 'non-functional testing' in software testing? Provide examples of each.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:189]:
```suggestion
      "What is the difference between 'functional testing' and 'non-functional testing' in software testing?",
```

File: src/data/quality-assurance-quiz.ts [hunk 6]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Answer:
+      "'Functional testing' assesses the functionality of a software application by testing individual functions or features. Examples include unit testing and integration testing. 'Non-functional testing' evaluates non-functional aspects like performance, security, and usability. Examples include load testing, security testing, and usability testing.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:191]: answer is to long. Long explanations work better for the explanation field instead of the answer field. 

```suggestion
      "'Functional testing' assesses the functionality of a software application while  'Non-functional testing' evaluates non-functional aspects like performance, security, and usability.",
```

File: src/data/quality-assurance-quiz.ts [hunk 7]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Distractor1:
+      "Functional testing and non-functional testing are the same thing.",
+    Distractor2:
+      "Functional testing only involves manual testing, while non-functional testing uses automation.",
+    Distractor3: "Non-functional testing is limited to web applications.",
+    Explanation:
+      "Functional testing ensures that the software meets specified requirements, while non-functional testing assesses how well the software performs under various conditions.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:198]:
```suggestion
      "Functional testing' assesses the functionality of a software application by testing individual functions or features. Examples include unit testing and integration testing. 'Non-functional testing' evaluates non-functional aspects like performance, security, and usability. Examples include load testing, security testing, and usability testing..",
```

File: src/data/quality-assurance-quiz.ts [hunk 8]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Link: ""
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:199]: the link field should always be filled out

```suggestion
    Link: "https://www.freecodecamp.org/news/software-testing-beginners-guide/"
```

File: src/data/quality-assurance-quiz.ts [hunk 9]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+      "Explain the concept of 'Test Driven Development (TDD)' in software development. What are its key principles and benefits?",
+    Answer:
+      "'Test Driven Development (TDD)' is a development approach where developers write test cases before writing the actual code. The key principles include writing failing tests first, writing the minimum code required to pass the test, and continuously refactoring code. Benefits include improved code quality, faster development, and reduced debugging time.",
+    Distractor1: "TDD involves writing tests after the code is developed.",
+    Distractor2: "TDD is only applicable to front-end development.",
+    Distractor3: "TDD is primarily used for load testing.",
+    Explanation:
+      "TDD helps ensure that code is thoroughly tested and that new changes do not break existing functionality.",
+    Link: ""
+  },
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:212]: we already have a few similar questions in the file about TDD so let's remove this one

File: src/data/quality-assurance-quiz.ts [hunk 10]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+  {
+    Question:
+      "What is 'Exploratory Testing,' and how does it differ from scripted testing? When is it most useful?",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:215]:
```suggestion
      "What is 'Exploratory Testing,' and how does it differ from scripted testing?",
```

File: src/data/quality-assurance-quiz.ts [hunk 11]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Answer:
+      "'Exploratory Testing' is a testing approach where testers explore the software application without predefined test cases. They use their domain knowledge and experience to find defects and assess the software's usability. It differs from scripted testing, where tests are preplanned and documented. Exploratory testing is most useful for uncovering unexpected issues and assessing user experience.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:217]: answer is to long. Would work better in the explanation field instead of the answer field

```suggestion
      "'Exploratory Testing' is a testing approach where testers explore the software application without predefined test cases. It differs from scripted testing, where tests are preplanned and documented.",
```

File: src/data/quality-assurance-quiz.ts [hunk 12]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Distractor1: "Exploratory testing is primarily automated.",
+    Distractor2:
+      "Scripted testing is less structured than exploratory testing.",
+    Distractor3: "Exploratory testing is only used for security testing.",
+    Explanation:
+      "Exploratory testing is valuable for finding defects that scripted testing might miss and for evaluating the software from a user's perspective.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:223]:
```suggestion
      "'Exploratory Testing' is a testing approach where testers explore the software application without predefined test cases. They use their domain knowledge and experience to find defects and assess the software's usability. It differs from scripted testing, where tests are preplanned and documented. Exploratory testing is most useful for uncovering unexpected issues and assessing user experience.",
```

File: src/data/quality-assurance-quiz.ts [hunk 13]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Link: ""
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:224]: link field should never be blank

```suggestion
    Link: "https://www.freecodecamp.org/news/software-testing-beginners-guide/"
```

File: src/data/quality-assurance-quiz.ts [hunk 14]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+  },
+  {
+    Question:
+      "What is 'Continuous Integration (CI)' and 'Continuous Delivery (CD)' in the context of software testing and deployment? How do they improve software quality and release processes?",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:228]:
```suggestion
      "What is 'Continuous Integration (CI)' and 'Continuous Delivery (CD)' in the context of software testing and deployment?",
```

File: src/data/quality-assurance-quiz.ts [hunk 15]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Answer:
+      "'Continuous Integration (CI)' is the practice of frequently merging code changes into a shared repository and automatically running tests to detect integration issues. 'Continuous Delivery (CD)' extends CI by automating the deployment process, allowing for rapid and reliable software releases. They improve software quality by catching issues early and enabling faster, more reliable releases.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:230]: answer is to long. Long answers work better for the explanation field. 

```suggestion
      "'CI/CD is an automated process that involves frequent code integration, automated testing, and continuous deployment of software changes to production.",
```

File: src/data/quality-assurance-quiz.ts [hunk 16]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Distractor1: "CI and CD are the same thing.",
+    Distractor2: "CI and CD are only relevant for mobile app development.",
+    Distractor3: "CI and CD focus on manual testing exclusively.",
+    Explanation:
+      "CI and CD promote a culture of automation and collaboration, reducing the risk of defects and accelerating the delivery of new features to users.",
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:235]:
```suggestion
      "'Continuous Integration (CI)' is the practice of frequently merging code changes into a shared repository and automatically running tests to detect integration issues. 'Continuous Delivery (CD)' extends CI by automating the deployment process, allowing for rapid and reliable software releases. They improve software quality by catching issues early and enabling faster, more reliable releases.",
```

File: src/data/quality-assurance-quiz.ts [hunk 17]
```diff
@@ -182,6 +182,58 @@ const qualityAssuranceQuiz = [
...
+    Link: ""
```
Comment (jdwilkin4) [inline src/data/quality-assurance-quiz.ts:236]: link field should not be blank

```suggestion
    Link: "https://www.freecodecamp.org/news/what-is-ci-cd/#how-ci-cd-works"
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍
